Application/EdFi.Ods.Tests/EdFi.Ods.Common/Security/AutoUpgradingHashedSecretVerifierDecoratorTests.cs

```diff
@@ -496,8 +498,17 @@ protected override void Arrange()
 
                 _apiClientSecretProvider = Stub<IApiClientSecretProvider>();
 
-                var secureHasher = new Pbkdf2HmacSha1SecureHasher(null);
-                var next = new SecureHashAwareSecretVerifier(packedHashConverter, secureHasher);
+                var secureHasher = new Pbkdf2HmacSha1SecureHasher();
+                var originalHasher = A.Fake<ISecureHasher>();
+                A.CallTo(() => originalHasher.GetAlgorithmHashCode).Returns(originalHashAlgorithm);
+                A.CallTo(() => originalHasher.ComputeHash(A<string>._, A<int>._, A<int>._, A<byte[]>._)).Returns(packedHash);
+
+                var next = new SecureHashAwareSecretVerifier(
```

Comment:
Should the class being decorated just be faked?  Seems stranger we're using full implementations all the way through for a unit test.

Reply:
i intially thought about going that way, but instead i decided that i wanted to validate it was working correctly as I stepped through the code so i added the concrete class.